Assets/Scripts/Foliage/SpawnFoliage.cs

```diff
@@ -1,161 +1,247 @@
-using System.Collections;
-using System.Collections.Generic;
-using Unity.Burst.CompilerServices;
-using Unity.VisualScripting;
 using UnityEngine;
+using System;
+using Random = UnityEngine.Random;
+using System.Linq;
 
 
 /// <summary>
 /// Script which spawns given foliage on a planet
 /// </summary>
 
 public class SpawnFoliage : MonoBehaviour
-{ 
-
-    // Temporary while testing
-    [SerializeField] private float rayBallRadius;
-    [SerializeField] private float planetRadius;
-    [SerializeField] private float rayLenght;
-    [SerializeField] private float spawningAngleLimit;
-    [SerializeField] private int stoneLimit;
-    [SerializeField] private int bushLimit;
-    [SerializeField] private int treeLimit;
-    [SerializeField] private int seed;
-    [SerializeField] private float treeLine;
-
-    [SerializeField] private Object[] trees = new Object[4];
-    [SerializeField] private Object[] bushes = new Object[2];
-    //[SerializeField] private Object[] grass = new Object[2];
-    [SerializeField] private Object[] stones = new Object[2];
-
-    // Start is called before the first frame update
-    void Start()
-    {
-        Random.InitState(seed);
+{
+    const int prefabLimit = 10000;
 
-        plantStones();
-        plantBushes();
-        plantTrees();
-    }
+    [Header("Trees")]
+    [SerializeField] private int treeLimit = 1000;
+    [SerializeField] private float treeLine = 25;
+    [SerializeField] private float treeAngleLimit = 30;
+    [SerializeField] private GameObject[] treePrefabs = new GameObject[4];
 
-    /// <summary>
-    /// Plants trees
-    /// </summary>
-    private void plantTrees()
+    [Header("Bushes")]
+    [SerializeField] private int bushLimit = 5000;
+    [SerializeField] private float bushLine = 35;
+    [SerializeField] private float bushAngleLimit = 30;
+    [SerializeField] private GameObject[] bushPrefab = new GameObject[4];
+
+    [Header("Rocks")]
+    [SerializeField] private int stoneLimit = 400;
+    [SerializeField] private float stoneLine = 36;
+    [SerializeField] private float stoneAngleLimit = 30;
+    [SerializeField] private GameObject[] stonePrefab = new GameObject[4];
+
+    [Header("Misc")]
+    [SerializeField] private int seed = 0;
```

Comment:
Maybe adjust so it uses the universal seed from the Universe script

Reply:
+

Reply:
Fixat